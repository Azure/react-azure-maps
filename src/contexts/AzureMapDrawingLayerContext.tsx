import React, { createContext } from 'react'
import { useAzureMapDrawingLayer } from '../hooks/useAzureMapDrawingLayer'
import { IAzureDrawingLayerStatefulProviderProps, IAzureMapLayerProps } from '../types'

const AzureMapDrawingLayerContext = createContext<IAzureMapLayerProps>({
  layerRef: null
})
const { Provider, Consumer: AzureMapDrawingLayerConsumer } = AzureMapDrawingLayerContext

const AzureMapDrawingLayerStatefulProvider = ({
  options,
  type,
  events,
  lifecycleEvents
}: IAzureDrawingLayerStatefulProviderProps) => {
  const { layerRef } = useAzureMapDrawingLayer({
    options,
    type,
    events,
    lifecycleEvents
  })

  return (
    <Provider
      value={{
        layerRef
      }}
    ></Provider>
  )
}

export {
  AzureMapDrawingLayerContext,
  AzureMapDrawingLayerConsumer,
  AzureMapDrawingLayerStatefulProvider as AzureMapDrawingLayerProvider
}
