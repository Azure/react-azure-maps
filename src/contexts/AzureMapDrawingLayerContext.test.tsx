import React from 'react'
import { render } from '@testing-library/react'
import { AzureMapDrawingLayerProvider } from './AzureMapDrawingLayerContext'

describe('AzureMapDrawingLayerProvider', () => {
  it('should create and render AzureMapDrawingLayerProvider', () => {
    const { container } = render(<AzureMapDrawingLayerProvider type="polygonLayer" options={{fillColor: 'green'}}/>)
    expect(container).toMatchSnapshot()
  })
})
