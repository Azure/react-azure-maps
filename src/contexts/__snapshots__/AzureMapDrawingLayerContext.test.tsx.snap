// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AzureMapDrawingLayerProvider should create and render AzureMapDrawingLayerProvider 1`] = `<div />`;
